### ISSUE TITLE
[Feature Request] easy way to test with files

### ISSUE THREAD

**Issue by demarey** [guillep/pharo-core#11790]
In Pharo, there is no easy way to do test cases dealing with files without writing code dealing with file cleaning.
I propose to define a subclass of TestCase that will provide a way to create temporary files and directories on disk, ensuring their cleaning after the test run. It will also provide a method to get a MemoryFileSystem quickly.

Here is the proposition:
```smalltalk
TestCase subclass: #FileTestCase
	instanceVariableNames: 'directories files'
	classVariableNames: ''
	package: 'SUnit-Core-Kernel'!

!FileTestCase methodsFor: 'creating'!
newMemoryFileSystem 
	
	"no need to clean it, it is a memory file system"

	^ FileSystem memory root! !

!FileTestCase methodsFor: 'creating'!
newTempDirectory

	| directory |
	
	directory := (FileLocator temp / (self class name,  '-' , UUIDGenerator next asString)) resolve.
	directory ensureCreateDirectory.
	directories add: directory.
	^ directory! !

!FileTestCase methodsFor: 'creating'!
newTempFile

	| file |
	
	file := (FileLocator temp / (self class name,  '-' , UUIDGenerator next asString)) resolve.
	files add: file.
	^ file! !


!FileTestCase methodsFor: 'initialization'!
initialize 
	
	directories := OrderedCollection new.
	files := OrderedCollection new.! !


!FileTestCase methodsFor: 'running'!
tearDown 

	directories do: [ :dir | dir ensureDeleteAll ].
	files do: [ :file | file ensureDelete ]! !

```

**Comment by Ducasse** [comment]
Nice we have already some tests of fileSystemTests doing the same but for memory file. 
So may be this one should be call
PhysicalFileRootTestCase

**Comment by demarey** [comment]
@Ducasse I cannot see a class like this one in FileSystemTests. For memory file system, cleaning is useless as files will be deleted once the memory file system is garbage collected.
The goal of this proposition is to have a ready-to-use class you can use as super class of your test class when you need to create temporary files.

**Comment by Ducasse** [comment]
Yes I wanted to say that it would be good to have a common superclass of tests that use FileSystem memory.

**Comment by tinchodias** [comment]
Did you consider these alternatives without subclassing?

## alternative 1:
The code you proposed but in a class that subclasses Object, and TestCases instance it and use it as a collaborator (stored in an instance variable). Like a tradicional test suite resource but whose lifecycle is a test execution.

## alternative 2:
The code you proposed but as stateful trait

**Comment by demarey** [comment]
I finally came back to this issue with a super simple approach. I added 2 methods on TestCase
- `memoryDirectory` that returns a new memory filesystem root. You need to keep a reference to it in your tests to use it
- `temporaryDirectory` that returns a file reference in the system temporary location (with a specific UUID). This file reference is deleted in the tear down if the method has been used

**Comment by tinchodias** [comment]
Alexandrie & Bloc have a related requirement: read files in their Iceberg repository.

Concretely, for the case of Alexandrie:
- https://github.com/pharo-graphics/Alexandrie/blob/a8fbed3df88f302535e82bbe207ab6be80a0af40/src/Alexandrie-Base-Tests/AeFilesystemResources.class.st provides the filereference to the directory with the required .PNGs and fonts.
- A trick is needed in smalltalkci: Add `#registerInIceberg : true` in the `.ston` definition (https://github.com/pharo-graphics/Alexandrie/blob/a8fbed3df88f302535e82bbe207ab6be80a0af40/.ci.ston#L7C7-L7C32). Otherwise, this fails in the github workflow, as by default the code of your project is loaded without adding it to iceberg, then the `AeFilesystemResources` class doesn't find the repo in Iceberg singleton.



<blockquote><img src="https://opengraph.githubassets.com/68086dd262bb538910f201d2b6c13195d1eff2ca7f76198800d8b547f9f1d0f8/pharo-graphics/Alexandrie" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/pharo-graphics/Alexandrie/blob/a8fbed3df88f302535e82bbe207ab6be80a0af40/src/Alexandrie-Base-Tests/AeFilesystemResources.class.st">Alexandrie/src/Alexandrie-Base-Tests/AeFilesystemResources.class.st at a8fbed3df88f302535e82bbe207ab6be80a0af40 · pharo-graphics/Alexandrie</a></strong></div><div>FFI bindings and a 2D canvas for Pharo based on Cairo, Freetype and Harfbuzz - pharo-graphics/Alexandrie</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/68086dd262bb538910f201d2b6c13195d1eff2ca7f76198800d8b547f9f1d0f8/pharo-graphics/Alexandrie" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/pharo-graphics/Alexandrie/blob/a8fbed3df88f302535e82bbe207ab6be80a0af40/.ci.ston">Alexandrie/.ci.ston at a8fbed3df88f302535e82bbe207ab6be80a0af40 · pharo-graphics/Alexandrie</a></strong></div><div>FFI bindings and a 2D canvas for Pharo based on Cairo, Freetype and Harfbuzz - pharo-graphics/Alexandrie</div></blockquote>

**Comment by jecisc** [comment]
If you need to access files from the git repository maybe this can help you: https://github.com/jecisc/GitBridge

There is a documentation in the readme and also we have this blogpost for a usage in moose: https://modularmoose.org/blog/2025-10-08-testing-your-algo-on-a-java-project/

<blockquote><img src="https://opengraph.githubassets.com/a0536079b3e61b97000e5effc3399216baf49f65a560a00724566d8649062952/jecisc/GitBridge" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/jecisc/GitBridge">GitHub - jecisc/GitBridge: Access resources and information from the git repository containing your project.</a></strong></div><div>Access resources and information from the git repository containing your project. - jecisc/GitBridge</div></blockquote>
<blockquote><img src="/favicon.svg" width="48" align="right"><div>Modular Moose</div><div><strong><a href="https://modularmoose.org/blog/2025-10-08-testing-your-algo-on-a-java-project/">Testing your algo on a java project</a></strong></div></blockquote>

**Comment by tinchodias** [comment]
I like GitBridge, thanks @jecisc I wasn't aware of it.

**Comment by jecisc** [comment]
If you check the code you'll see that it is really simple, yet useful. With Stephane we wondered once if this could not be integrated in Iceberg directly
